src/main/kotlin/gg/skytils/skytilsmod/listeners/DungeonListener.kt

```diff
@@ -222,11 +241,15 @@ object DungeonListener {
                                     entranceLoc = entrance.mainRoom.z * entrance.mainRoom.x
                                 ))
                                 while (DungeonTimer.dungeonStartTime != -1L) {
-                                    while (outboundRoomQueue.isNotEmpty()) {
-                                        val packet = outboundRoomQueue.poll() ?: continue
+                                    for (packet in outboundRoomQueue) {
```

Comment:
I'm not sure about the underlying implementation of iterating over a channel but is the issue really that `isNotEmpty` is not constant time? i.e. is this suspending rather than blocking when waiting for a new element?

Reply:
> I'm not sure about the underlying implementation of iterating over a channel but is the issue really that `isNotEmpty` is not constant time? i.e. is this suspending rather than blocking when waiting for a new element?

iterating over a channel should suspend the coroutine while it waits for a new element